redbot/core/__init__.py

```diff
@@ -1,11 +1,40 @@
-import pkg_resources
-
 from .config import Config
 from .context import RedContext
 
 __all__ = ["Config", "RedContext", "__version__"]
 
-try:
-    __version__ = pkg_resources.get_distribution("Red-DiscordBot").version
-except pkg_resources.DistributionNotFound:
-    __version__ = "3.0.0"
+
+class VersionInfo:
+    def __init__(self, major, minor, micro, releaselevel, serial):
+        self._levels = ['alpha', 'beta', 'final']
+        self.major = major
+        self.minor = minor
+        self.micro = micro
+
+        if releaselevel not in self._levels:
+            raise TypeError("'releaselevel' must be one of: {}".format(
+                ', '.join(self._levels)
+            ))
+
+        self.releaselevel = releaselevel
+        self.serial = serial
+
+    def __lt__(self, other):
+        my_index = self._levels.index(self.releaselevel)
+        other_index = self._levels.index(other.releaselevel)
+        return self.major < other.major or \
+            self.minor < other.minor or \
+            self.micro < other.micro or \
+            my_index < other_index or \
+            self.serial < other.serial
```

Comment:
This should compare as 
```py
return (self.major, self.minor, self.micro, my_index, self.serial) < (other.major, other.minor, other.micro, other_index, other.serial)
```
The way you have it here with nested or would have 3.0.1 be < 3.1.0

Reply:
3.0.1 is less than 3.1.0?

Reply:
my mistake, I meant that both of these version comparisons evaluate true by the above:

3.0.1 < 3.1.0
3.1.0 < 3.0.1